app/src/contexts/telemetryTableContext.tsx

```diff
@@ -150,77 +168,236 @@ export const TelemetryTableContextProvider = (props: ITelemetryTableContextProvi
+        updatedVisibilityModel = Object.fromEntries(tableColumns.map((column) => [column, !areAllColumnsVisible]));
       }
+      setColumnVisibilityModel(updatedVisibilityModel);
+    },
+    [columnVisibilityModel, getColumns, setColumnVisibilityModel]
+  );
+
+  /**
+   * Update edit or view state of table rows
+   *
+   * @param {string[]} rowIds - Row ids to modify
+   * @param {GirdRowModes} mode - Mode to change to ie: View or Edit
+   * @param {boolean} keepChanges - Indicator to keep or remove changes
+   * @returns {void}
+   */
+  const _updateRowsMode = useCallback(
+    (rowIds: string[], mode: GridRowModes, keepChanges: boolean) => {
+      setRowModesModel((prevModel) => {
+        const newModel: GridRowModesModel = { ...prevModel };
+        for (const id of rowIds) {
+          newModel[id] = { mode, ignoreModifications: !keepChanges };
+        }
+        return newModel;
+      });
+    },
+    [setRowModesModel]
+  );
 
-      return Object.entries(editRow).reduce(
-        (newRow, entry) => ({ ...row, ...newRow, _isModified: true, [entry[0]]: entry[1].value }),
-        {}
+  /**
+   * Checks if the cell has error.
+   *
+   * @returns {boolean}
+   */
+  const hasError = useCallback(
+    (params: GridCellParams): boolean => {
+      return Boolean(
+        validationModel[params.row.id]?.some((error) => {
+          return error.field === params.field;
+        })
       );
-    }) as IManualTelemetryTableRow[];
-  }, [_muiDataGridApiRef]);
+    },
+    [validationModel]
+  );
+
+  /**
+   * Puts the specified row into edit mode, and adds the row id to the array of modified rows.
+   *
+   * @param {GridRowId} id
+   */
+  const onRowEditStart = useCallback((id: GridRowId) => {
+    // Add row to modified rows array
+    _modifiedRowIds.current = Array.from(new Set([..._modifiedRowIds.current, String(id)]));
+  }, []);
 
   /**
-   * Returns all columns belonging to thte telemetry table.
+   * Refresh the telemetry records and pre-parse to table date format
+   *
+   * @async
+   * @returns {Promise<void>}
    */
-  const getColumns = useCallback(() => {
-    return _muiDataGridApiRef.current.getAllColumns?.() ?? [];
-  }, [_muiDataGridApiRef]);
+  const refreshRecords = useCallback(async () => {
+    const telemetry = (await telemetryDataContext.telemetryDataLoader.refresh(deployment_ids)) ?? [];
+
+    // Format the rows to use date and time
+    const rows: IManualTelemetryTableRow[] = telemetry.map((item) => {
+      return {
+        id: item.id,
+        deployment_id: item.deployment_id,
+        latitude: item.latitude,
+        longitude: item.longitude,
+        date: dayjs(item.acquisition_date).format('YYYY-MM-DD'),
+        time: dayjs(item.acquisition_date).format('HH:mm:ss'),
+        telemetry_type: item.telemetry_type
+      };
+    });
+
+    // Set initial rows for the table context
+    setRows(rows);
+  }, [deployment_ids]);
```

Comment:
Dependency warning. Probably just needs an ignore and maybe a comment about which dependency specifically is being ignored